plugins/chain_plugin/test/test_chain_plugin.cpp

```diff
@@ -0,0 +1,526 @@
+#define BOOST_TEST_MODULE chain_plugin
+#include <eosio/chain/permission_object.hpp>
+#include <boost/test/unit_test.hpp>
+#include <eosio/testing/tester.hpp>
+#include <eosio/chain/types.hpp>
+#include <eosio/chain_plugin/chain_plugin.hpp>
+#include <eosio/chain/abi_serializer.hpp>
+#include <fc/variant_object.hpp>
+#include <contracts.hpp>
+#include <eosio/chain/contract_table_objects.hpp>
+#include <eosio/chain/resource_limits.hpp>
+#include <eosio/chain/wast_to_wasm.hpp>
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <fc/log/logger.hpp>
+#include <eosio/chain/exceptions.hpp>
+#include <Runtime/Runtime.h>
+
+#ifdef NON_VALIDATING_TEST
+#define TESTER tester
+#else
+#define TESTER validating_tester
+#endif
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::chain_apis;
+using namespace eosio::testing;
+using namespace eosio::chain_apis;
+using namespace fc;
+
+using mvo = fc::mutable_variant_object;
+
+class chain_plugin_tester : public TESTER {
+public:
+
+    action_result push_action( const account_name& signer, const action_name &name, const variant_object &data, bool auth = true ) {
+         string action_type_name = abi_ser.get_action_type(name);
+
+         action act;
+         act.account = config::system_account_name;
+         act.name = name;
+         act.data = abi_ser.variant_to_binary( action_type_name, data, abi_serializer::create_yield_function(abi_serializer_max_time) );
+
+         return base_tester::push_action( std::move(act), (auth ? signer : signer == "bob111111111"_n ? "alice1111111"_n : "bob111111111"_n).to_uint64_t() );
+    }
+
+    action_result deposit( const account_name& owner, const asset& amount ) {
+      return push_action( name(owner), "deposit"_n, mvo()
+                          ("owner",  owner)
+                          ("amount", amount)
+                          );
+    }
+
+    void transfer( name from, name to, const asset& amount, name manager = config::system_account_name ) {
+      base_tester::push_action( "eosio.token"_n, "transfer"_n, manager, mutable_variant_object()
+                                ("from",    from)
+                                ("to",      to )
+                                ("quantity", amount)
+                                ("memo", "")
+                                );
+    }
+
+    action_result stake( const account_name& from, const account_name& to, const asset& net, const asset& cpu ) {
+      return push_action( name(from), "delegatebw"_n, mvo()
+                          ("from",     from)
+                          ("receiver", to)
+                          ("stake_net_quantity", net)
+                          ("stake_cpu_quantity", cpu)
+                          ("transfer", 0 )
+      );
+    }
+    action_result stake( std::string_view from, std::string_view to, const asset& net, const asset& cpu ) {
```

Comment:
Delete this unused overloaded stake() method.

Reply:
Fixed.